src/network/msgpackobject.cc

```diff
@@ -380,7 +380,7 @@ std::shared_ptr<MsgpackObject> toMsgpackObject(const double val) {
 
 void fromMsgpackObject(const std::shared_ptr<MsgpackObject> obj, std::shared_ptr<data::NodeID>& out) {
```

Comment:
The only solution for templating return type I could think of looks something like that (due limitations of being unable to do partial function specialization):

```
template <class T>
struct FromMsgpackObject
{
  static T call(const std::shared_ptr<messages::MsgpackObject> obj);
};
template <>
struct FromMsgpackObject<bool>
{
  static bool call(const std::shared_ptr<messages::MsgpackObject> obj) {return true;}
};
template <class S>
struct FromMsgpackObject<std::shared_ptr<std::vector<S>>>
{
  static std::shared_ptr<std::vector<S>> call(const std::shared_ptr<messages::MsgpackObject> obj) {return nullptr;}
};
```
and calling it like this:
```
FromMsgpackObject<bool>::call(nullptr);
FromMsgpackObject<std::shared_ptr<std::vector<bool>>>::call(nullptr);
```
I didn't think this looks better than current version, but I'm open to suggestions.